src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java

```diff
@@ -0,0 +1,47 @@
+package org.jenkinsci.plugins.displayurlapi.actions;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+import hudson.model.Action;
+import org.jenkinsci.plugins.displayurlapi.ClassicDisplayURLProvider;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public abstract class AbstractDisplayAction implements Action {
+
+    public static final String URL_NAME = "display";
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
+
+    @Override
+    public String getUrlName() {
+        return URL_NAME;
+    }
+
+    public final Object doRedirect(StaplerRequest req, StaplerResponse rsp) throws IOException {
```

Comment:
Not really, you can still move the code over and keep a `this.doIndex(…)` invocation for URL compatibility.